### PR TITLE
add flow definition file for the element API

### PR DESCRIPTION
**Pull request by koddsson**
https://github.com/github/web-systems/issues/215

### CODE REVIEW

File: index.js.flow [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/* @flow strict */
+
+declare module '@github/tab-container-element' {
+  declare export default class TabContainerElement extends HTMLElement {
+    selectTab(index: number): void;
```
Comment (muan) [inline index.js.flow]: I know this isn't a new method, but this solidifies it (by allowing `github/github` to cast the type and use this method 😬 ). 

IMO if this method is ever needed, things might have gone terribly wrong elsewhere. This should be easily achieved with `tabs[i].click()`. 

I should have reviewed this before #4. 💀 🤦‍♀

Comment (koddsson) [reply index.js.flow]: Hahaha no worries. I figured getting these APIs exposed is going to surface stuff like this and we can just deal with them in the PR's and bump the major version.

Is the terribly wrong scenario that we might be listening for tab clicks and by doing a `TabContainerElement.selectTab(1)` we are not dispatching a `click` event for those handlers?

Also should `tabs` be a property on the element or were you thinking users would do something like `const tabs = container.querySelectorAll('[role="tab"]')` in application code?

Comment (muan) [reply index.js.flow]: > Is the terribly wrong scenario that we might be listening for tab clicks and by doing a `TabContainerElement.selectTab(1)` we are not dispatching a `click` event for those handlers?

I think it is unlikely, and exposing this capability might encourage a bad pattern. What would be a valid case for needing a behavior _that is usually triggered by click_ happen without click? 

In most cases I don't think selecting tab based on `index` is something that people would want/need to do. For example in our write/preview tabs, we might want to open the preview programmatically, but in that case we shouldn't do `container.selectTab(1)`, we should do `previewTab.click()` to be explicit. 

> Also should `tabs` be a property on the element or were you thinking users would do something like `const tabs = container.querySelectorAll('[role="tab"]')` in application code?

`tabs[i].click()` was an example workaround for if people _really_ want to do this, which I don't think is common and shouldn't be encouraged. 😬

Comment (koddsson) [reply index.js.flow]: > > Is the terribly wrong scenario that we might be listening for tab clicks and by doing a `TabContainerElement.selectTab(1)` we are not dispatching a `click` event for those handlers?
> 
> I think it is unlikely, and exposing this capability might encourage a bad pattern. What would be a valid case for needing a behavior _that is usually triggered by click_ happen without click?

I'm not advocating for us keeping the method, just trying to clarify so I understand better. So I don't have a valid case 😄

**Pull request merged**